### PR TITLE
Global Path Service

### PR DESCRIPTION
**Pull request by SPDonaghy**
### Description

- Part of [#49 Mock Global Path](https://github.com/UBCSailbot/local_pathfinding/issues/49) (Local Pathfinding)

I implemented a Global Path Service message.
The request is a Path type and the response is a bool type. Given how infrequently the global path is expected to change, the choice was made to switch global path from a topic to a service. This way, nodes such as MockGlobalPath and LocalTransceiver  (which have access to the global path) can send a "request" message containing the new global path to Navigate. Then Navigate can respond with True to acknowledge receipt of the message.

### Verification

To verify my changes, I first ran the command "ros2 interface show custom_interfaces/srv/GlobalPath" and received the response:

Path global_path
       HelperLatLon[] waypoints
                float32 latitude
                float32 longitude

bool response

So the service was built properly.

I then launched the local pathfinding package and the mock global path node (which will use this service) ran as expected, along with the navigate node.

### Resources

- [Tutorial 2](https://docs.ros.org/en/foxy/Tutorials/Beginner-Client-Libraries/Custom-ROS2-Interfaces.html)
- [Tutorial 1](https://wiki.ros.org/ROS/Tutorials/CreatingMsgAndSrv)

### CODE REVIEW

File: README.md
Comment (patrick-5546) [inline README.md:1]: Let's make the type column the filename. For example, `LPathData.msg`

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Formatting reminder:
 | `gps`                  | GPS             | CanTrxRosIntf      | Controller, Local Transceiver, Local Pathfinding   |
 | `mock_gps`             | GPS             | Boat Simulator     | CanSimIntf                                         |
 | `local_path_data`      | LPathData       | Local Pathfinding  | Local Transceiver                                  |
-| `global_path`          | Path            | Local Transceiver  | Local Pathfinding                                  |
+| `global_path`(srv)     | Path <=> Bool   | LocalTrans, MGPath | Local Pathfinding                                  |
```
Comment (patrick-5546) [inline README.md]: Don't abbreviate local transceiver, name should be consistent across the table

File: srv/GlobalPathService.srv
Comment (patrick-5546) [inline srv/GlobalPathService.srv:1]: Rename to `GlobalPath.srv`

**Review by patrick-5546 — COMMENTED** [review]
@hhenry01 FYI we're changing the global path topic from msg to srv

File: README.md
Comment (patrick-5546) [reply README.md:1]: If you did this it was not pushed

File: README.md (hunk 1 above)
Comment (patrick-5546) [reply README.md]: Removed MGPath

File: README.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ Formatting reminder:
 | `gps`                  | GPS             | CanTrxRosIntf      | Controller, Local Transceiver, Local Pathfinding   |
 | `mock_gps`             | GPS             | Boat Simulator     | CanSimIntf                                         |
 | `local_path_data`      | LPathData       | Local Pathfinding  | Local Transceiver                                  |
-| `global_path`          | Path            | Local Transceiver  | Local Pathfinding                                  |
+| `global_path`(srv)     | GlobalPath      | Local Transceiver  | Local Pathfinding                                  |
```
Comment (patrick-5546) [inline README.md]: Removed MGPath

File: README.md
Comment (SPDonaghy) [reply README.md:1]: Oh I understand, you want me to change all of them and change the column name to 'filename' ?

Comment (patrick-5546) [reply README.md:1]: Yes that would be great

**Review by patrick-5546 — APPROVED** [review]

**Comment by jamenkaye** [comment]
Looks good, nice work Sean!

**Pull request merged**